### PR TITLE
ibmi: uv_resident_set_memory() returns 0

### PR DESCRIPTION
**Pull request by dmabupt**
On IBM i we can not get the resident set memory size of a process. 
Current code does not support **yarn** --
> yarn version
Error: ENOSYS: function not implemented, uv_resident_set_memory

So return 0 just like what the **ps** command does on IBM i and make **yarn** work.

### CODE REVIEW

File: src/unix/ibmi.c [hunk 1]
```diff
@@ -66,13 +66,13 @@ uint64_t uv_get_total_memory(void) {
 
 
 void uv_loadavg(double avg[3]) {
-    avg[0] = avg[1] = avg[2] = 0;
-    return;
+  avg[0] = avg[1] = avg[2] = 0;
```
Comment (cjihrig) [inline src/unix/ibmi.c]: The indentation changes are unrelated. Can you revert them.

**Comment by gireeshpunathil** [comment]
LGTM. Does it make sense to update to the doc to this effect as applicable?

**Comment by gireeshpunathil** [comment]
btw, the `grey octocat` symbol near your commit message is an indication that your terminal did not have your `git config user.id` set up. I guess pushing next change with the right config will fix it.

**Comment by kadler** [comment]
Shouldn't we set `rss` to *some* value (0 seems fine to me)? Otherwise, the caller will think whatever random data was in the buffer passed is the actual rss, which could lead to unexpected behavior.

**Comment by ThePrez** [comment]
LGTM with the latest commit. @mgoffmn is the last @libuv/ibmi reviewer

**Comment by ThePrez** [comment]
Agree with @kadler. We need code to set `rss` to 0.

**Comment by mgoffmn** [comment]
LGTM

**Comment by gireeshpunathil** [comment]
I am not sure I follow the new change. Setting rss with 0 is good, but can't we still `return UV_ENOSYS;` ? So that the API consumers get the right picture?

**Comment by dmabupt** [comment]
@gireeshpunathil Returning UV_ENOSYS makes the tool **yarn** crash.
```
yarn versions
Error: ENOSYS: function not implemented, uv_resident_set_memory
    at process.memoryUsage (internal/process.js:103:5)
    at ConsoleReporter.checkPeakMemory (/QOpenSys/pkgs/lib/nodejs10/lib/node_modules/yarn/lib/cli.js:36397:40)
    at ConsoleReporter.initPeakMemoryCounter (/QOpenSys/pkgs/lib/nodejs10/lib/node_modules/yarn/lib/cli.js:36388:10)
    at /QOpenSys/pkgs/lib/nodejs10/lib/node_modules/yarn/lib/cli.js:83808:14
    at Generator.next (<anonymous>)
    at step (/QOpenSys/pkgs/lib/nodejs10/lib/node_modules/yarn/lib/cli.js:92:30)
    at /QOpenSys/pkgs/lib/nodejs10/lib/node_modules/yarn/lib/cli.js:110:14
    at new Promise (<anonymous>)
    at new F (/QOpenSys/pkgs/lib/nodejs10/lib/node_modules/yarn/lib/cli.js:23624:28)
    at /QOpenSys/pkgs/lib/nodejs10/lib/node_modules/yarn/lib/cli.js:89:12
```

**Comment by santigimeno** [comment]
I'm still confused... shouldn't the `UV_ENOSYS` error be handled by node itself?

**Comment by cjihrig** [comment]
I think Node is doing the right thing by throwing when it sees `UV_ENOSYS` in `process.memoryUsage()`. I also think the most correct behavior is for libuv to return `UV_ENOSYS` here, but IBM i seems to be a bit of a special snowflake here.

**Comment by santigimeno** [comment]
> I think Node is doing the right thing by throwing when it sees UV_ENOSYS in process.memoryUsage()

Couldn't node just set the `rss` field to `0` (or `undefined`) in the `UV_ENOSYS` case as the other fields of the returned object could be calculated?

**Comment by cjihrig** [comment]
Yea, it could. I think that is just pushing the issue downstream though. I think errors in `process.memoryUsage()` should be somewhat rare, but should result in thrown exceptions. Pushing this to Node would make it ignore errors on all platforms instead of dealing with it here for just IBM i.

**Comment by santigimeno** [comment]
I understand your point though I would argue that this issue is specifically downstream and returning `UV_ENOSYS` looks like the correct thing to me. Also I was only referring to the `UV_ENOSYS` error which has very specific semantics and is returned by `ibm i` and `cygwin` and not to every other error.

**Comment by cjihrig** [comment]
Just my opinion. I don't have really strong feelings either way, as I'm not an IBM i or cygwin user.

**Comment by santigimeno** [comment]
Let's see what others think, I won't be blocking this if it's considered a good change.

**Comment by dmabupt** [comment]
In my opinion, this change only affects IBM i users and makes yarn installed correctly at least. We may implement the function in future. But for now, it does not make things worse.

**Comment by aaronbartell** [comment]
Another use case that has the `uv_resident_set_memory` error...

All of the [good* npms](https://www.npmjs.com/package/good) (good, good-console, good-file, good-squeeze; all related to hapijs) fail with this same error.

```
{ Error: ENOSYS: function not implemented, uv_resident_set_memory
    at memoryUsage (internal/process.js:103:5)
    at Promise (/home/aaron/git/web/node_modules/oppsy/lib/utils.js:9:59)
    at new Promise (<anonymous>)
    at Object.psmem (/home/aaron/git/web/node_modules/oppsy/lib/utils.js:7:16)
    at Timeout._interval.setInterval [as _onTimeout] (/home/aaron/git/web/node_modules/oppsy/lib/index.js:42:49)
    at ontimeout (timers.js:498:11)
    at tryOnTimeout (timers.js:323:5)
    at Timer.listOnTimeout (timers.js:290:5)
  errno: -109,
  code: 'ENOSYS',
  syscall: 'uv_resident_set_memory' }
```

**Comment by aaronbartell** [comment]
Another use case is when using the [good*](https://www.npmjs.com/package/good) npms (a common HapiJs logging means).

```
{ Error: ENOSYS: function not implemented, uv_resident_set_memory
    at memoryUsage (internal/process.js:103:5)
    at Promise (/home/aaron/git/biaero_web/node_modules/oppsy/lib/utils.js:9:59)
    at new Promise (<anonymous>)
    at Object.psmem (/home/aaron/git/biaero_web/node_modules/oppsy/lib/utils.js:7:16)
    at Timeout._interval.setInterval [as _onTimeout] (/home/aaron/git/biaero_web/node_modules/oppsy/lib/index.js:42:49)
    at ontimeout (timers.js:498:11)
    at tryOnTimeout (timers.js:323:5)
    at Timer.listOnTimeout (timers.js:290:5)
  errno: -109,
  code: 'ENOSYS',
  syscall: 'uv_resident_set_memory' }
```

## Steps

```
npm i hapijs good good-console good-file good-squeeze
```

```js
'use strict';
const Hapi   = require('hapi');

const options = {
  ops: {
      interval: 1000
  },
  reporters: {
    myFileReporter: [{
        module: 'good-squeeze',
        name: 'Squeeze',
        args: [{ ops: '*' }]
    }, {
        module: 'good-squeeze',
        name: 'SafeJson'
    }, {
        module: 'good-file', 
        args: ['./test/fixtures/awesome_log']
    }]
  }
};

const server = Hapi.server({
  port: 8000,
  host: '0.0.0.0'
});

const init = async () => {
  await server.register({ plugin: require('good'), options });
  await server.start();
  console.log(`Server running at: ${server.info.uri}`);
};

process.on('unhandledRejection', (err) => {
  process.exit(1);
});

init();
```

**Comment by cjihrig** [comment]
@santigimeno are you OK with me landing this?

**Comment by santigimeno** [comment]
Sure. Maybe it's worth making the same change in `https://github.com/libuv/libuv/blob/v1.x/src/unix/cygwin.c#L41` before landing.

**Comment by cjihrig** [comment]
@santigimeno I just proposed that as a separate change: https://github.com/libuv/libuv/pull/1992

**Comment by cjihrig** [comment]
CI was OK: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/1028/

Landed in https://github.com/libuv/libuv/commit/8813dca3883bb808e33e947c8784ffe16792ca94. Thanks!
